### ISSUE TITLE
Plugin is not correctly extracting the basepath

### ISSUE THREAD

**Issue by vuksv** [srinandan/bprouter#2]
When using this plugin, if make a request to something like /sample/additional-url-path
the plugin will make a request to the lookupEndpoint with _sample_additional-url-path

This behavior means that any dynamic URL will not properly get looked up in the KVM which tries to match the basepath

**Comment by vuksv** [comment]
My current fix is to use the base path configured in the proxy

```javascript
var basePath = res.proxy.base_path;
```

instead of

```javascript
var basePath = url.parse(req.url).pathname;
```

It seems like this should be the required behavior. Is there an issue with this approach?

**Comment by zycon** [comment]
@vuksv 
Yep. in  the case proxy base path and target basepath is different, it will throw a 404,
try this
` req.targetPath = res.proxy.parsedUrl.path;`
